src/Reader/Reader.php

```diff
@@ -189,17 +197,14 @@ public static function useHttpConditionalGet($bool = true)
      */
     public static function import($uri, $etag = null, $lastModified = null)
     {
-        $cache       = self::getCache();
-        $client      = self::getHttpClient();
-        $client->resetParameters();
-        $headers = new ZendHttp\Headers();
-        $client->setHeaders($headers);
-        $client->setUri($uri);
+        $cache   = self::getCache();
+        $client  = self::getHttpClient();
         $cacheId = 'Zend_Feed_Reader_' . md5($uri);
 
         if (static::$httpConditionalGet && $cache) {
-            $data = $cache->getItem($cacheId);
-            if ($data) {
+            $headers = [];
+            $data    = $cache->getItem($cacheId);
+            if ($data && $client instanceof Http\HeaderAwareClientInterface) {
```

Comment:
The reason this is here is because there's no reason to aggregate the headers if the client is not capable of using them anyways. Skipping the code inside the conditional allows skipping 2 cache fetch operations.